### PR TITLE
Add support for country selection based on MCC

### PR DESCRIPTION
**Pull request by Herrie82**
Add support for country selection based on MCC

### CODE REVIEW

File: qml/CountryPage.qml [hunk 1]
```diff
@@ -26,8 +28,77 @@ BasePage {
     forwardButtonSourceComponent: forwardButton
 
     property variant currentRegion: null
+    property string mccCountryCode: ""
     property int currentRegionIndex: -1
 
+    Item {
```
Comment (morphis) [inline qml/CountryPage.qml]: This should be better a separate component NetworkIdCountryMapper.qml

File: qml/CountryPage.qml [hunk 2]
```diff
@@ -26,8 +28,77 @@ BasePage {
...
+            } else {
+                __currentCall = connectNetwork.subscribe(JSON.stringify({
+                                                                            subscribe: true
+                                                                        }))
+            }
+        }
+    }
+
+    LunaService {
+        id: connectNetwork
```
Comment (morphis) [inline qml/CountryPage.qml]: connectNetwork -> queryNetworkId

File: qml/CountryPage.qml [hunk 3]
```diff
@@ -26,8 +28,77 @@ BasePage {
...
+            var response = JSON.parse(message.payload)
+            var mcc = parseInt(response.extended.mccmnc.substring(0, 3))
+            for (var n = 0; n < dataModel.count; n++) {
+                var entry = dataModel.get(n)
+                if (mcc === entry.mcc) {
+                    mccCountryCode = entry.CountryCode.toLowerCase()
+                    console.log("We found countrycode: "+mccCountryCode+" for MCC: "+response.extended.mccmnc.substring(0, 3))
+                }
+            }
+            getRegions()
```
Comment (morphis) [inline qml/CountryPage.qml]: When this a separate component just send a signal here "done" or assign a property which the outside can then listen for.

File: qml/CountryPage.qml [hunk 4]
```diff
@@ -26,8 +28,77 @@ BasePage {
...
+            handleUpdatedModel()
+        }
+
+        property variant __currentCall: null
+
+        function handleUpdatedModel() {
+            if (__currentCall) {
+                __currentCall.cancel()
+            } else {
+                __currentCall = connectNetwork.subscribe(JSON.stringify({
```
Comment (morphis) [inline qml/CountryPage.qml]: connectNetwork -> queryNetworkId

File: qml/CountryPage.qml [hunk 5]
```diff
@@ -52,55 +123,74 @@ BasePage {
...
+                "countryCode": countryCode
             }
-        };
+        }
+
+        setPreferences.call(JSON.stringify(request))
+    }
 
-        setPreferences.call(JSON.stringify(request));
+    function getRegions() {
```
Comment (morphis) [inline qml/CountryPage.qml]: getRegions -> retrieveRegions

Only method which directly return something are prefixed with "get".

**Comment by Herrie82** [comment]
Updated as requested

**Comment by Herrie82** [comment]
Added support for timezone selection now as well. Based on MCC it will determine the country + offset from mccInfo.json and will find the exact country + offset match or will find the closest based on the difference in offset values. Test for countries with a single timezone + multiple timezones - and + from GMT. All seem to work. Might needs a bit of refactoring still, but working.

File: qml/main.qml [hunk 6]
```diff
@@ -42,6 +42,7 @@ Window {
     }
 
     StackView {
+        property int mccOffsetFromUTC: -1
```
Comment (morphis) [inline qml/main.qml]: Why do we have this here? If you need to share things between two components use a static js library.

Comment (Herrie82) [inline qml/main.qml]: How do I do this?

I also thought it was out of place here, but had no idea how to solve 
this differently.

**Pull request closed without merging**